go/store/types/map_test.go

```diff
@@ -2109,3 +2109,34 @@ func TestMapWithNil(t *testing.T) {
 		NewSet(context.Background(), vrw, String("a"), String("b"), Float(42), nil)
 	})
 }
+
+func TestVisitMapLevelOrderSized(t *testing.T) {
+	smallTestChunks()
+	defer normalProductionChunks()
+
+	assert := assert.New(t)
+	vrw := newTestValueStore()
+	kvs := []Value{}
+	for i := 0; i < testMapSize; i++ {
+		kvs = append(kvs, Float(i), Float(i+1))
+	}
+
+	m, err := NewMap(context.Background(), vrw, kvs...)
+	d.PanicIfError(err)
+
+	expectedChunkHashes := make([]hash.Hash, 0)
+	_, _, err = VisitMapLevelOrder(m, func(h hash.Hash) (int64, error) {
+		expectedChunkHashes = append(expectedChunkHashes, h)
+		return 0, nil
+	})
+	d.PanicIfError(err)
+
+	actualChunkHashes := make([]hash.Hash, 0)
+	_, _, err = VisitMapLevelOrderSized([]Map{m}, 37624, func(h hash.Hash) (int64, error) {
+		actualChunkHashes = append(actualChunkHashes, h)
+		return 0, nil
+	})
+	d.PanicIfError(err)
+
+	assert.Equal(expectedChunkHashes, actualChunkHashes)
```

Comment:
Might want to sort these, and add another case where you test `VisitMapLevelOrderSized` with a real small limit like `2`, just to make sure everything is proceeding as expected.